### ISSUE TITLE
Model broken when applying NNCF as a wrapper. Output tensors have 'inf' as values,

### ISSUE THREAD

**Issue by CristianPopov** [openvinotoolkit/nncf#873]
Hello,

I have stumbled upon a bug when applying NNCF to a certain model and I'd need your help with this issue.

**Target**: quantize model https://github.com/gengshan-y/high-res-stereo

**Issue**: when applying NNCF (even as a wrapper, without any compression algorithm), the resulting model is broken. The output tensors have all the values 'inf'.

**Background**:
Due to some issues with ModelOptimizer, some changes need to be made in the environment to be able to do all the exports.
To do a proper PyTorch -> ONNX export, I had to manually revert the change in torch/onnx/symbolic_opset11.py from https://github.com/pytorch/pytorch/commit/3c5e3966f4a05edbf959888fe42dbfe89d746571 , to have a ONNX model that does not have any 'If_' nodes in the graph. ModelOptimizer does not handle 'If_' nodes at the moment.
Another chagne I had to do was to manually apply https://github.com/openvinotoolkit/openvino/pull/6739 into the installed OpenVino (I can't wait for the next OpenVino release which will include this fix). Without this one, ModelOptimizer will break the IR model when doing the ONNX -> IR conversion, and the output tensors will be empty (all zero values).
Another thing to note here is that this model cannot be loaded into OpenVino in ONNX format. OpenVino is not there in terms of support.
The last thing to mention is that this model is huge, if you were to check the graph. Sorry about this, but I do not have a smaller model to reproduce this issue, to ease the debug.

In FP32 I have extracted the accuracy numbers (average error being the most important) from the PyTorch, ONNX and IR models, and all of the models gave me the same average error. This way I am sure that all the conversions are working.

With OpenVino POT I also get a working IR model, although the accuracy is broken (4-5x average error) so I wanted to try with NNCF to compare accuracy numbers, since I made it work with NNCF in the past on a different model, but with NNCF applied I can't get a working IR model.

**Steps to reproduce the issue:**
1. Train the model applying NNCF wrapper (no compression algorithm):
python3 train_nncf_reduced_dataset.py --batch_size 2 --database /path/to/extracted/dataset/ --save_model nncf_checkpoint/ --load_model hsm_fp32_model.tar --epochs 11  
Training log attached: [nncf_train.log](https://github.com/openvinotoolkit/nncf/files/6891171/nncf_train.log)

2. Copy the model into the root folder, to continue with the export:
cp nncf_checkpoint/nncf_log1/finetune_29.tar nncf_model.tar

3. Convert the PyTorch NNCF wrapped checkpoint into an ONNX model (takes about 5 minutes)
python3 nncf_convert_to_onnx.py --database /path/to//extracted/dataset/ 
Conversion log attached:  [nncf_convert_to_onnx.log](https://github.com/openvinotoolkit/nncf/files/6891392/nncf_convert_to_onnx.log)

4. Convert ONNX model to IR: (takes up to 20 minutes)
/home/cdpopov/intel/openvino_2021.3.394/deployment_tools/model_optimizer/mo.py --input_model hsm_nncf.onnx
Conversion log attached: [convert_onnx_to_ir.log](https://github.com/openvinotoolkit/nncf/files/6891689/convert_onnx_to_ir.log)

5. Run submission script (1st step in getting the accuracy numbers):
python3 submission_openvino_nncf.py --data_path /path/to/extracted/dataset/mb-ex-training/trainingF/ --out_dir benchmark_nncf_ov/stage3/

/path/to/extracted/dataset/mb-ex-training/trainingF//Adirondack-imperfectL/im0.png
[[[inf inf inf ... inf inf inf]
  [inf inf inf ... inf inf inf]
  [inf inf inf ... inf inf inf]
  ...
  [inf inf inf ... inf inf inf]
  [inf inf inf ... inf inf inf]
  [inf inf inf ... inf inf inf]]]
Total inference time (ms)= 1552.64
CPU usage(%):  632.9
Here's where I get the broken output.

If you want to do the same for the clean FP32 model : PyTorch -> ONNX -> IR -> OpenVino inference to get accuracy numbers, for comparison, do the following:
python3 convert_to_onnx.py
/home/cdpopov/intel/openvino_2021.3.394/deployment_tools/model_optimizer/mo.py --input_model hsm_fp32_384.onnx
python3 submission_openvino.py --data_path /path/to/extracted/dataset/mb-ex-training/trainingF/ --out_dir benchmark_ov/stage3/
python3 eval_mb_simple.py --in_dir benchmark_ov/stage3/ --gt_dir /path/to/extracted/dataset/mb-ex/trainingF/ --file benchmark_ov/stage3.txt
cat benchmark_ov/stage3.txt
[average_error.log](https://github.com/openvinotoolkit/nncf/files/6892077/average_error.log)

Environment:
OS: Ubuntu 18.04.5, kernel 5.4.0-66-generic
CPU: Intel(R) Core(TM) i9-10980XE CPU @ 3.00GHz
OpenVino: 2021.3.394
virtual env:  [python_env.txt](https://github.com/openvinotoolkit/nncf/files/6891716/python_env.txt)

All the scripts and models from the steps above and the code that should be worked on is available here: https://intel-my.sharepoint.com/:u:/p/cristian_d_popov/EauVreNAtLhGodDaVMs-RV4BUwpl4Cb8Qqv-aAHIHgbqNA?email=alexander.kozlov%40intel.com&e=ixdt6C . Due to known confidentiality politics, I can't share this publicly. If you need me to share this to a different developer, let me know and I will create new share links.

**Comment by AlexKoff88** [comment]
@asenina, can you please take a look?

**Comment by CristianPopov** [comment]
https://intel-my.sharepoint.com/:u:/p/cristian_d_popov/EauVreNAtLhGodDaVMs-RV4BUwpl4Cb8Qqv-aAHIHgbqNA?email=anastasia.senina%40intel.com&e=e0TXWD share link for @asenina

**Comment by CristianPopov** [comment]
Hi guys,

Is there any feedback that could be shared for this issue? I know that you are busy and that this bug is low priority for you, which is fine, but I would appreciate any kind of feedback on this issue.

**Comment by asenina** [comment]
Hi, @CristianPopov!
Sorry for the long waiting. I have already started to solve this issue.

**Comment by CristianPopov** [comment]
I have been contacted by @asenina  in private, through our company's channels and found out were the issue was. One of the flags from my scripts was triggering the issue, when applying NNCF as a wrapper. After changing it to the correct value everything is working and I can now apply NNCF quantization to my model.

These scripts I am using were mostly inherited and I did not check every line of them, since I did not observe any issue without NNCF in the picture.
Sorry for wasting your time.
Thank you.
